components/brave_wallet/browser/internal/hd_key_zip32.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_ZIP32_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_ZIP32_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "brave/components/brave_wallet/common/buildflags.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+#include "brave/components/brave_wallet/zcash/lib.rs.h"
+
+static_assert(BUILDFLAG(ENABLE_ORCHARD));
+
+namespace brave_wallet {
+
+enum class OrchardKind {
+  // External kind, can be used in account addresses
+  External,
+  // Internal "change" address
+  Internal
+};
+
+// Implements Orchard key generation from
+// https://zips.z.cash/zip-0032#orchard-child-key-derivation
+class HDKeyZip32 {
+ public:
+  explicit HDKeyZip32(rust::Box<OrchardExtendedSpendingKeyResult> esk);
+  HDKeyZip32(const HDKeyZip32&) = delete;
+  HDKeyZip32& operator=(const HDKeyZip32&) = delete;
+
+  ~HDKeyZip32();
+
+  // Generates master key using provided seed
+  static std::unique_ptr<HDKeyZip32> GenerateFromSeed(
+      base::span<const uint8_t> seed);
+
+  // Derives hardened key using index and the current key
```

Comment:
what is a "hardened key"? Comments should provide details that don't assume the person reading them knows anything about zcash.

Reply:
Hardened key is a non-zcash specific term that is widely used in crypto, not sure if we need to comment on such things. cc @darkdh , @yrliou 
https://gist.github.com/arnabmitra/b9eeddc5356550216abfcc9d5571fe41

Reply:
That is right, it is a [bip32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) term when talking about Hierarchical Deterministic Wallets and it is encapsulated in `brave_wallet::HDKey`

Reply:
so put that in the comments?

Reply:
I guess it's fine